sdks/python/apache_beam/runners/portability/fn_api_runner_test.py

```diff
@@ -801,6 +807,48 @@ def contains_labels(monitoring_info, labels):
       print(res._monitoring_infos_by_stage)
       raise
 
+  def test_duplicated_callbacks(self):
+    event_recorder = EventRecorder(tempfile.gettempdir())
+    event_recorder.record('test')
+    elements_list = ['1', '2']
+
+    class FinalizebleDoFnWithSameRegister(beam.DoFn):
+      def process(
+          self,
+          element,
+          bundle_finalizer=beam.DoFn.BundleFinalizerParam):
+        bundle_finalizer.register(lambda: event_recorder.delete_file())
```

Comment:
Are you trying to test here that a failing finalizer does not fail the pipeline? If so, make that explicit via a finializer that raises an exception.